src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -16822,7 +16822,814 @@ public class Helper
+delegate TResult MyFunc<in T1,in T2,out TResult>(T1 arg1, T2 arg2)
+    where T1 : allows ref struct
+    where T2 : allows ref struct
+    where TResult : allows ref struct;
+
+interface I1
+{
+    int P {get;}
+}
+
+ref struct S1
+{
+    public int P {get;set;}
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // Errors are expected for both queries, but it is a pre-existing condition - https://github.com/dotnet/roslyn/issues/72945    
+                );
+        }
+
+        [Fact]
+        public void Makeref()
+        {
+            var src = @"
+public class Helper
+{
+    static void Test1<T>(T x)
+        where T : allows ref struct
+    {
+        System.TypedReference tr = __makeref(x);
+    }
+
+    static void Test2(S y)
+    {
+        System.TypedReference tr = __makeref(y);
+    }
+}
+
+ref struct S
+{
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (7,36): error CS1601: Cannot make reference to variable of type 'T'
+                //         System.TypedReference tr = __makeref(x);
+                Diagnostic(ErrorCode.ERR_MethodArgCantBeRefAny, "__makeref(x)").WithArguments("T").WithLocation(7, 36),
+                // (12,36): error CS1601: Cannot make reference to variable of type 'S'
+                //         System.TypedReference tr = __makeref(y);
+                Diagnostic(ErrorCode.ERR_MethodArgCantBeRefAny, "__makeref(y)").WithArguments("S").WithLocation(12, 36)
+                );
+        }
+
+        [Fact]
+        public void ScopedTypeParameter_01()
+        {
+            var src = @"
+public class Helper
+{
+    static void Test1<T>(scoped T x)
+        where T : allows ref struct
+    {
+    }
+
+    static void Test2(scoped S y)
+    {
+    }
+
+    static void Test3<T>(scoped T z)
+    {
+    }
+
+    static void Test4<T>()
+    {
+        var d = void (scoped T u) => {};   
+        d(default);
+    }
+}
+
+ref struct S
+{
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (13,26): error CS9048: The 'scoped' modifier can be used for refs and ref struct values only.
+                //     static void Test3<T>(scoped T z)
+                Diagnostic(ErrorCode.ERR_ScopedRefAndRefStructOnly, "scoped T z").WithLocation(13, 26),
+                // (19,23): error CS9048: The 'scoped' modifier can be used for refs and ref struct values only.
+                //         var d = void (scoped T u) => {};   
+                Diagnostic(ErrorCode.ERR_ScopedRefAndRefStructOnly, "scoped T u").WithLocation(19, 23)
+                );
+
+            var src2 = @"
+public class Helper
+{
+    public static void Test1<T>(scoped T x)
```

Comment:
Consider testing `scoped ref T` parameter as well.

Reply:
> Consider testing `scoped ref T` parameter as well.

Why do you think this is interesting? Any `ref` parameter can be declared scoped, no type restrictions for that.